consensus/misc/basefee.go

```diff
@@ -0,0 +1,133 @@
+	"github.com/ethereum/go-ethereum/params"
+)
+
+var (
+	errInvalidBaseFee       = errors.New("invalid BaseFee")
+	errMissingParentBaseFee = errors.New("parent header is missing BaseFee")
+	errMissingBaseFee       = errors.New("current header is missing BaseFee")
+	errHaveBaseFee          = errors.New("BaseFee should not be set before EIP1559 activation %d")
+)
+
+// VerifyEIP1559BaseFee verifies that the EIP1559 BaseFee field is valid for the current block height
+func VerifyEIP1559BaseFee(config *params.ChainConfig, header, parent *types.Header) error {
+	if config.IsEIP1559(parent.Number) {
+		if parent.BaseFee == nil {
+			return errMissingParentBaseFee
+		}
+	}
+	expectedBaseFee := CalcBaseFee(config, parent)
+	if header.BaseFee == nil {
+		if expectedBaseFee != nil {
+			return errMissingBaseFee
+		}
+		return nil
+	}
+	if expectedBaseFee == nil {
+		if header.BaseFee != nil {
+			return errHaveBaseFee
+		}
+		return nil
+	}
+	if header.BaseFee.Cmp(expectedBaseFee) != 0 {
+		return errInvalidBaseFee
+	}
+	return nil
+}
+
+func computeBaseFee(pBaseFee *big.Int, pGasUsed, pGasTarget, denominator uint64) *big.Int {
+	var baseFee *big.Int
+	if pGasUsed == pGasTarget {
+		baseFee = new(big.Int).Set(pBaseFee)
+	} else if pGasUsed > pGasTarget {
+		gasDelta := big.NewInt(int64(pGasUsed) - int64(pGasTarget))
+		feeDelta := math.BigMax(
+			new(big.Int).Div(
+				new(big.Int).Mul(pBaseFee, gasDelta),
+				new(big.Int).Mul(
+					new(big.Int).SetUint64(pGasTarget),
+					new(big.Int).SetUint64(denominator),
+				),
+			),
+			big.NewInt(1),
+		)
+		baseFee = new(big.Int).Add(pBaseFee, feeDelta)
+	} else {
+		gasDelta := big.NewInt(int64(pGasTarget) - int64(pGasUsed))
+		feeDelta := new(big.Int).Div(
+			new(big.Int).Mul(pBaseFee, gasDelta),
+			new(big.Int).Mul(
+				new(big.Int).SetUint64(pGasTarget),
+				new(big.Int).SetUint64(denominator),
+			),
+		)
+		baseFee = new(big.Int).Sub(pBaseFee, feeDelta)
+	}
+	return baseFee
+}
+
+// CalcBaseFee returns the baseFee for the current block provided the parent header and config parameters
+func CalcBaseFee(config *params.ChainConfig, parent *types.Header) *big.Int {
+	height := new(big.Int).Add(parent.Number, common.Big1)
+
+	// If we are before EIP1559 activation, the baseFee is nil
+	if !config.IsEIP1559(height) {
+		return nil
+	}
+
+	// If we are at the block of EIP1559 activation then the BaseFee is set to the initial value
+	if config.EIP1559Block.Cmp(height) == 0 {
+		return new(big.Int).SetUint64(config.EIP1559.InitialBaseFee)
+	}
+
+	return computeBaseFee(
+		parent.BaseFee,
+		parent.GasUsed,
+		parent.GasLimit,
+		config.EIP1559.EIP1559BaseFeeMaxChangeDenominator,
+	)
+}
+
+// CalcEIP1559GasTarget returns the EIP1559GasTarget at the current height and header.GasLimit
+// This should only be called at or above the block height of EIP1559 activation and below finalization
+func CalcEIP1559GasTarget(chainConfig *params.ChainConfig, height, gasLimit *big.Int) *big.Int {
```

Comment:
Do we still need this method ? Since we don't have the migration period anymore  i think we don't need it